### PR TITLE
fix quantity extension

### PR DESCRIPTION
**Pull request by maxperei**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?   | no

fix type error when trying to get product id from object instead of array

### CODE REVIEW

File: src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php [hunk 1]
```diff
@@ -51,27 +43,30 @@ public function getFunctions()
         ];
     }
 
-    public function getProductQuantity($product): int
+    public function getProductQuantity(mixed $product): int
```
Comment (mi-lopez) [inline src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php:46]: mixed D:< 

now we can add the possible types like:

`public function getProductQuantity(array|Product $product): int`

File: src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php [hunk 2]
```diff
@@ -51,27 +43,30 @@ public function getFunctions()
...
+        $product = $this->getProductObj($product);
         $stockAlert = $this->stockAlertDataProvider->getStockAlertForProduct($product);
         if ($stockAlert instanceof StockAlert) {
             return true;
         }
         return false;
     }
 
-    protected function getProductObjec($product) : Product
+    protected function getProductObj(mixed $product): Product
```
Comment (mi-lopez) [inline src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php:62]: `    protected function getProductObj(Product|ProductView|array $product): Product`

File: src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php (hunk 1 above)
Comment (mi-lopez) [reply src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php:46]: well... mixed is also enough I guess xD

**Review by mi-lopez — APPROVED** [review]

File: src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php (hunk 1 above)
Comment (maxperei) [reply src/Synolia/Bundle/StockAlertBundle/Twig/QuantityExtension.php:46]: i'm not sure but there are possibly 3 types you can pass via template imports ; `Product|ProductView|array` so it's very confusing

**Pull request merged**